Yoast/javascript
# Update the Twitter preview component

## Summary
There was an existing TwitterPreview component, however it's style did not reflect the actual Twitter Preview. This PR mainly updates styling so that the component looks more like the actual Twitter Preview.


This PR can be summarized in the following changelog entry:

* [social-metadata-previews] Reactify the TwitterPreview component and make it up-to-date with the actual Twitter Preview.

## Important
This PR also fixes #157 because it felt counterproductive to add the truncation function without being able to see the final result. The first commit contains the changes for the `TwitterPreview` component, the second commit contains the changes for the truncation.

## Relevant technical choices:
* I tried to use flexbox instead of boxes and float.
* The SVG icon is copied from Twitter.
* Styles have been copied from Twitter as well.
* Description and title will not be cropped, this will be done in https://github.com/Yoast/javascript/issues/157 (I have a feeling about who is going to pick that up...)

## Test instructions

Currently, the component is not used by the plugin. This PR will only be merged in the feature repository. Therefore, the following test suffices:

This PR can be tested by following these steps:
* Check out the component in the Example Components app.
* Find a preview on Twitter
* Check that they look alike.
* Also check different preview types.

## UI changes (although this will be merged in a `feature` branch)

## Quality assurance


Fixes #48 
Fixes #157
